src/test/groovy/hpo/api/HpoLoincServiceSpec.groovy

```diff
@@ -0,0 +1,60 @@
+package hpo.api
+
+import grails.testing.services.ServiceUnitTest
+import hpo.api.util.Loinc2Hpo
+import hpo.api.util.Loinc2HpoFactory
+import org.monarchinitiative.loinc2hpo.loinc.LoincEntry
+import org.monarchinitiative.loinc2hpo.loinc.LoincId
+import org.monarchinitiative.phenol.ontology.data.TermId
+import spock.lang.Shared
+import spock.lang.Specification
+
+class HpoLoincServiceSpec extends Specification implements ServiceUnitTest<HpoLoincService> {
+
+  @Shared
+  Loinc2Hpo loinc2Hpo
+
+  void setupSpec() {
+    loinc2Hpo = new Loinc2HpoFactory().getInstance()
+  }
+
+  void setup(){
+    service.hpoLoinc = loinc2Hpo
+  }
+
+  void "search loinc by hpo"() {
+
+    given:
+      TermId query = TermId.of("HP:0004363")
+      Map<LoincId, LoincEntry> mockLoincEntryMap = new HashMap<>()
+      Map<TermId, Set<LoincId>> mockReverseMap = new HashMap<>()
+
+      mockLoincEntryMap.put(buildMockLoincId("38230-9"), buildMockLoincEntry(getOneLoincLine()))
+      mockReverseMap.put(TermId.of("HP:0004363"), new HashSet<LoincId>(Arrays.asList(new LoincId("38230-9"))))
+      service.hpoLoinc = Mock(Loinc2Hpo)
+      service.hpoLoinc.getReverseAnnotationMap() >> mockReverseMap
+      service.hpoLoinc.getLoincEntryMap() >> mockLoincEntryMap
```

Comment:
when method gets called return the mock

Reply:
It seems setupSpec() tries to read all annotation and loincEntries anyway. Should we also mock Loinc2Hpo?

Reply:
This is true. On my first pass I didn't look at the code you have commented out. So I just implemented the whole thing.. Then I used a test level mock here. The reading of all annotation files part can probably be deleted. I was just doing it to show how you would do it.

Reply:
I deleted it now.